### PR TITLE
Refactor 303 quarkus vertx sql module

### PR DESCRIPTION
**Pull request by Sgitario**
I'm planning to implement more scenarios in regards of Route validation:
- Add a custom validator
- Validate responses

Beforehand, this PR aims to make some cleanup with:
- Rename test classes in order to avoid having duplicate class names in different packages
- Remove unnecessary @Valid and validator annotations
- Fix some warnings that my IDE was complaining about

### CODE REVIEW

File: 303-quarkus-vertx-sql/src/main/java/io/quarkus/qe/vertx/sql/services/FlightSearchService.java [hunk 1]
```diff
@@ -37,7 +39,7 @@ void onStart(@Observes StartupEvent ev) {
         return Flight.findByOriginDestination(connection, query.from, query.to)
                 .onItem()
                 .transform(flight -> calculatePrice(query, flight))
-                .collectItems().in(ArrayList::new, List::add);
+                .collect().in(ArrayList::new, List::add);
```
Comment (pjgg) [inline 303-quarkus-vertx-sql/src/main/java/io/quarkus/qe/vertx/sql/services/FlightSearchService.java:42]: why this you change this method?

File: 303-quarkus-vertx-sql/src/test/java/io/quarkus/qe/vertx/sql/handlers/airline/Db2AirlineHandlerTest.java [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
+package io.quarkus.qe.vertx.sql.handlers.airline;
 
 import io.quarkus.qe.vertx.sql.test.resources.Db2TestProfile;
 import io.quarkus.test.junit.QuarkusTest;
 import io.quarkus.test.junit.TestProfile;
 
 @QuarkusTest
 @TestProfile(Db2TestProfile.class)
-public class Db2HandlerTest extends AirlineHandlerTest {
+public class Db2AirlineHandlerTest extends AirlineHandlerTest {
```
Comment (pjgg) [inline 303-quarkus-vertx-sql/src/test/java/io/quarkus/qe/vertx/sql/handlers/airline/Db2AirlineHandlerTest.java:9]: Redundant name -> Db2HandlerTest  applies to this context `io.quarkus.qe.vertx.sql.handlers.airline`

Comment (pjgg) [reply 303-quarkus-vertx-sql/src/test/java/io/quarkus/qe/vertx/sql/handlers/airline/Db2AirlineHandlerTest.java:9]: this comment applies to all bellow changes.

File: 303-quarkus-vertx-sql/src/test/java/io/quarkus/qe/vertx/sql/test/resources/Db2Resource.java [hunk 3]
```diff
@@ -49,7 +52,7 @@ private void defaultDb2Container(Map<String, String> config) {
 
     @Override
     public void stop() {
-        if (Objects.nonNull(db2Container)) db2Container.stop();
+        Optional.ofNullable(db2Container).ifPresent(GenericContainer::stop);
```
Comment (pjgg) [inline 303-quarkus-vertx-sql/src/test/java/io/quarkus/qe/vertx/sql/test/resources/Db2Resource.java:55]: which is the additional value?

**Review by pjgg — CHANGES_REQUESTED** [review]
Well, I see that you apply Quarkus.io code style template! :) 

I don't see the point of adding the package name to the Test className. What are you looking for with this change?

File: 303-quarkus-vertx-sql/pom.xml [hunk 4]
```diff
@@ -54,7 +54,6 @@
         <dependency>
             <groupId>io.vertx</groupId>
             <artifactId>vertx-mongo-client</artifactId>
-            <version>4.0.0</version>
             <scope>test</scope>
         </dependency>
```
[the remark this answers is not in the document]
Comment (Sgitario) [reply 303-quarkus-vertx-sql/pom.xml:58]: Removed

File: 303-quarkus-vertx-sql/src/main/java/io/quarkus/qe/vertx/sql/services/FlightSearchService.java (hunk 1 above)
Comment (Sgitario) [reply 303-quarkus-vertx-sql/src/main/java/io/quarkus/qe/vertx/sql/services/FlightSearchService.java:42]: `.collectItems` is deprecated by `.collect`.

File: 303-quarkus-vertx-sql/src/test/java/io/quarkus/qe/vertx/sql/handlers/airline/Db2AirlineHandlerTest.java (hunk 2 above)
Comment (Sgitario) [reply 303-quarkus-vertx-sql/src/test/java/io/quarkus/qe/vertx/sql/handlers/airline/Db2AirlineHandlerTest.java:9]: This is to have unique classes names. Before, we had a few Db2HandlerTest classes. Having unique names is useful when troubleshoting test failures or search by a class name.

File: 303-quarkus-vertx-sql/src/test/java/io/quarkus/qe/vertx/sql/test/resources/Db2Resource.java (hunk 3 above)
Comment (Sgitario) [reply 303-quarkus-vertx-sql/src/test/java/io/quarkus/qe/vertx/sql/test/resources/Db2Resource.java:55]: Having `if` statements in a single line is generally a bad practice.

**Comment by Sgitario** [comment]
@pjgg I've replied all your comments and removed the leftover dependency, thanks for spotting!

**Review by pjgg — APPROVED** [review]

**Pull request merged**